source/array_list.c

```diff
@@ -24,6 +24,8 @@ int aws_array_list_init_dynamic(struct aws_array_list *list,
     size_t allocation_size = initial_item_allocation * item_size;
     list->data = NULL;
     list->item_size = item_size;
+    list->current_size = 0;
+    list->length = 0;
```

Comment:
a few lines below here length is set to 0. You can delete that line since you already set it above. GitHub won't let me highlight the offending line :|